extensions/data-workspace/src/dialogs/newProjectQuickpick.ts

```diff
@@ -0,0 +1,83 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+import * as constants from '../common/constants';
+import { directoryExist } from '../common/utils';
+import { defaultProjectSaveLocation } from '../common/projectLocationHelper';
+import { WorkspaceService } from '../services/workspaceService';
+
+/**
+ * Create flow for a New Project using only VS Code-native APIs such as QuickPick
+ */
+export async function createNewProjectWithQuickpick(workspaceService: WorkspaceService): Promise<void> {
+	// Refresh list of project types
+	const projectTypes = (await workspaceService.getAllProjectTypes()).map(projType => {
+		return {
+			label: projType.displayName,
+			description: projType.description,
+			id: projType.id
+		} as vscode.QuickPickItem & { id: string };
+	});
+
+	// 1. Prompt for project type
+	const projectType = await vscode.window.showQuickPick(projectTypes, { title: constants.SelectProjectType });
+	if (!projectType) {
+		return;
+	}
+
+	// 2. Prompt for project name
+	const projectName = await vscode.window.showInputBox(
```

Comment:
For the project name I would see ignoreFocusOut being enabled as beneficial, users clicking another screen to check the name they want to type here would cause the command to need to be restarted

Reply:
Would you have a problem with adding it for all the dialogs? I find it way too easy to accidently click off in general and when in a flow like this that would be really annoying.

Reply:
I would prefer it on all of them :)